src/MSIdentityScaffolding/Microsoft.DotNet.MSIdentity/MicrosoftIdentityPlatform/MicrosoftIdentityPlatformApplicationManager.cs

```diff
@@ -27,109 +28,106 @@ public class MicrosoftIdentityPlatformApplicationManager
-            {
-                await ExposeWebApiScopes(graphServiceClient, createdApplication, applicationParameters);
+                    // Re-reading the app to be sure to have everything.
+                    createdApplication = (await graphServiceClient.Applications
+                        .Request()
+                        .Filter($"appId eq '{createdApplication.AppId}'")
+                        .GetAsync()).FirstOrDefault();
+                }
 
-                // Re-reading the app to be sure to have everything.
-                createdApplication = (await graphServiceClient.Applications
-                    .Request()
-                    .Filter($"appId eq '{createdApplication.AppId}'")
-                    .GetAsync()).FirstOrDefault();
-            }
+                // log json console message inside this method since we need the Microsoft.Graph.Application
+                if (createdApplication is null)
+                {
+                    consoleLogger.LogFailure(Resources.FailedToCreateApp);
+                    Environment.Exit(1);
+                }
 
-            // log json console message inside this method since we need the Microsoft.Graph.Application
-            if (createdApplication is null)
-            {
-                consoleLogger.LogFailure(Resources.FailedToCreateApp, commandName);
-                return null;
-            }
+                if (applicationParameters.IsB2C)
+                {
+                    createdApplication.AdditionalData.Add("IsB2C", true);
+                }
 
-            if (applicationParameters.IsB2C)
-            {
-                createdApplication.AdditionalData.Add("IsB2C", true);
-            }
+                ApplicationParameters? effectiveApplicationParameters = GetEffectiveApplicationParameters(tenant!, createdApplication, applicationParameters);
 
-            ApplicationParameters? effectiveApplicationParameters = GetEffectiveApplicationParameters(tenant!, createdApplication, applicationParameters);
+                // Add password credentials
+                if (applicationParameters.CallsMicrosoftGraph || applicationParameters.CallsDownstreamApi)
+                {
+                    await AddPasswordCredentialsAsync(
+                        graphServiceClient,
+                        createdApplication.Id,
+                        effectiveApplicationParameters,
+                        consoleLogger);
+                }
+
+                var output = string.Format(Resources.CreatedAppRegistration, effectiveApplicationParameters.ApplicationDisplayName, effectiveApplicationParameters.ClientId);
+                consoleLogger.LogJsonMessage(State.Success, content: createdApplication, output: output);
 
-            // Add password credentials
-            if (applicationParameters.CallsMicrosoftGraph || applicationParameters.CallsDownstreamApi)
+                return effectiveApplicationParameters;
+            }
+            catch (Exception ex)
```

Comment:
later we should get some specific exceptions down, we can track this for now.

Reply:
Sounds good, added an issue to track